site/gatsby-site/src/utils/AlgoliaUpdater.js

```diff
@@ -302,78 +302,84 @@ class AlgoliaUpdater {
-        });
-
-        const dateSubmittedDescReplicaIndex = await this.algoliaClient.initIndex(
-          dateSubmittedDescReplicaIndexName
-        );
-
-        await dateSubmittedDescReplicaIndex.setSettings({
-          ranking: ['desc(epoch_date_submitted)'],
-        });
-
-        const dateSubmittedAscReplicaIndex = await this.algoliaClient.initIndex(
-          dateSubmittedAscReplicaIndexName
-        );
-
-        await dateSubmittedAscReplicaIndex.setSettings({
-          ranking: ['asc(epoch_date_submitted)'],
-        });
+    try {
+      await index.setSettings(
+        {
+          ...algoliaSettings,
+          indexLanguages: [language],
+          queryLanguages: [language],
+          replicas: [
+            featuredReplicaIndexName,
+            incidentDateDescReplicaIndexName,
+            incidentDateAscReplicaIndexName,
+            datePublishedDescReplicaIndexName,
+            datePublishedAscReplicaIndexName,
+            dateSubmittedDescReplicaIndexName,
+            dateSubmittedAscReplicaIndexName,
+          ],
+        },
+        {
+          forwardToReplicas: true,
+        }
+      );
+
+      const featuredReplicaIndex = await this.algoliaClient.initIndex(featuredReplicaIndexName);
+
+      await featuredReplicaIndex.setSettings({
+        ranking: ['desc(featured)', 'desc(text)'],
+      });
+
+      const incidentDateDescReplicaIndex = await this.algoliaClient.initIndex(
+        incidentDateDescReplicaIndexName
+      );
+
+      await incidentDateDescReplicaIndex.setSettings({
+        ranking: ['desc(epoch_incident_date)'],
       });
+
+      const incidentDateAscReplicaIndex = await this.algoliaClient.initIndex(
+        incidentDateAscReplicaIndexName
+      );
+
+      await incidentDateAscReplicaIndex.setSettings({
+        ranking: ['asc(epoch_incident_date)'],
+      });
+
+      const datePublishedDescReplicaIndex = await this.algoliaClient.initIndex(
+        datePublishedDescReplicaIndexName
+      );
+
+      await datePublishedDescReplicaIndex.setSettings({
+        ranking: ['desc(epoch_date_published)'],
+      });
+
+      const datePublishedAscReplicaIndex = await this.algoliaClient.initIndex(
+        datePublishedAscReplicaIndexName
+      );
+
+      await datePublishedAscReplicaIndex.setSettings({
+        ranking: ['asc(epoch_date_published)'],
+      });
+
+      const dateSubmittedDescReplicaIndex = await this.algoliaClient.initIndex(
+        dateSubmittedDescReplicaIndexName
+      );
+
+      await dateSubmittedDescReplicaIndex.setSettings({
+        ranking: ['desc(epoch_date_submitted)'],
+      });
+
+      const dateSubmittedAscReplicaIndex = await this.algoliaClient.initIndex(
+        dateSubmittedAscReplicaIndexName
+      );
+
+      await dateSubmittedAscReplicaIndex.setSettings({
+        ranking: ['asc(epoch_date_submitted)'],
+      });
+    } catch (e) {
+      console.log(e);
+      throw 'Error updating Algolia settings ' + e.message;
```

Comment:
Where this console.log is displayed? Is it in the Netlify build log?

Reply:
Yup, I'll remove it since the throw is enough.